src/mpi/misc/utils.c

```diff
@@ -26,6 +28,8 @@ int MPIR_Localcopy(const void *sendbuf, MPI_Aint sendcount, MPI_Datatype sendtyp
     sdata_sz = sendsize * sendcount;
     rdata_sz = recvsize * recvcount;
 
+    send_type = recv_type = MPL_GPU_POINTER_UNREGISTERED_HOST;
```

Comment:
Why is this initialization needed?

Reply:
We check for memory type when freeing the pack/unpack buffer at the end. This is to initialize the default case, i.e., pageable host memory

Reply:
Why is there a default case?  The query will return the right type, even if that's unregistered host.

Reply:
Right but we need to allocate memory only in the case in which both types are non contiguous, there is no need to query memory for other cases. So I would just set this manually to unregister host and query only if needed.

Reply:
But you are not using them in any other case, right?  They are only used in the noncontig case.

Reply:
If pack/unpack fails I still need to free the allocated buffer when exiting the function, so I need to know what buffer was allocated depending on the send/recv memory type.

Reply:
If the pack/unpack in the contig cases fail, there's no buffer to free.

Reply:
Yes. There are two options: keep everything local to the branch and in case of type mismatch clear the `buf` before `MPIR_ERR_CHKANDJUMP` or (as I do now) clear the buffer in `fn_fail`. The second approach is what we typically do.

Reply:
Alright, I think we spent more time arguing about it than what it deserves.  :-)  I'll ignore it.